Runtime/Rule/AnyLayerRule.cs

```diff
@@ -19,7 +19,7 @@ public class AnyLayerRule : GameObjectRule
         /// <inheritdoc />
         protected override bool Accepts(GameObject targetGameObject)
         {
-            return (targetGameObject.layer & LayerMask.value) != 0;
+            return LayerMask == (LayerMask | (1 << targetGameObject.layer));
```

Comment:
```suggestion
            return (LayerMask & (1 << targetGameObject.layer)) != 0;
```
This should also do it and makes more sense, at least to me 😄: "Does this mask contain any layer of this game object, turned into a mask (by `1 << layer`?" Feel free to ignore this suggestion.

Do we want to put this into an extension method for `LayerMask` to solve once and for all or do we think this will be the only usage going forward?

Reply:
I tried that (as it was in the other PR with a fix) but it still didn't work in actual practice for me

Reply:
Also, it's the only usage right now, we should probably move to an extension if it is used elsewhere down the line. No point in making an extension for a single use case.

Reply:
Merge ahead then.

Reply:
I do wonder why the alternative didn't work though. I figured that other person raised a PR with that as the fix, but then someone else said it didn't work for them in slack and it didn't work for me either.

Reply:
Maybe you were just missing parentheses or something? `(A & B) != 0` is definitely checking whether A "includes" B...

Alternatively remember that `GetMask` returns the mask (which is `1 << layerIndex`), `NameToLayer` returns just the layer's index (not a mask). Mask to int seems to be done via this: https://docs.unity3d.com/ScriptReference/LayerMask-value.html

Reply:
NameToLayer definitely didn't work in the tests. It was setting the LayerMask to `Default` + `Ignore RayCast` instead of `UI`

this is the code I literally copied and pasted:

https://github.com/ExtendRealityLtd/Zinnia.Unity/pull/379/commits/691eccf65ec56ffbfe5c8baa73228274aad52a4c

For the initial attempt at fixing it (when I was trying it in the scene) and it didn't seem to work with that code.

But I've just noticed that code isn't what you posted above:

Other PR -> `return (targetGameObject.layer & (1 << LayerMask.value)) != 0;`

Your Code -> `return (LayerMask & (1 << targetGameObject.layer)) != 0;`

Original Code -> `return (targetGameObject.layer & LayerMask.value) != 0;`

So maybe your suggestion will work. I'll try it tomorrow and report back.

Reply:
Yeah as far as I understand Unity's docs the `Other PR` basically is confusing mask vs. layer index (i.e. `.value`).

Mask == `1 << layerIndex`, with `someMask.value` being the index? Docs are sparse on that one.

Reply:
Ok so your original suggestion works and passes the new tests too.